service/worker/deployment/deployment_util.go

```diff
@@ -91,8 +96,7 @@ func ValidateDeploymentWfParams(fieldName string, field string, maxIDLengthLimit
 // EscapeChar is a helper which escapes the DeploymentWorkflowIDDelimeter character
 // in the input string
 func escapeChar(s string) string {
-	s = strings.Replace(s, `\`, `\\`, -1)
-	s = strings.Replace(s, DeploymentWorkflowIDDelimeter, `\`+DeploymentWorkflowIDDelimeter, -1)
+	s = strings.Replace(s, DeploymentWorkflowIDDelimeter, DeploymentWorkflowIDDelimeter+DeploymentWorkflowIDDelimeter, -1)
```

Comment:
good catch. I didn't think of the split issue. right, the scape should be different than the delimiter. ":" and "|" seem like good options.

Reply:
David had mentioned this in our earlier discussion and I seemed to have forgotten ( I should probably right things down cause its hard to remember these edge cases when discussing these in sync since we don't have much time to ponder upon)

thank you! changed!